sway/tree/workspace.c

```diff
@@ -454,3 +456,60 @@ bool workspace_is_empty(struct sway_container *ws) {
 	}
 	return true;
 }
+
+static int find_output(const void *id1, const void *id2) {
+	return strcmp(id1, id2) ? 0 : 1;
+}
+
+void workspace_output_raise_priority(struct sway_container *workspace,
+		struct sway_container *old_output, struct sway_container *output) {
+	struct sway_workspace *ws = workspace->sway_workspace;
+
+	int old_index = list_seq_find(ws->output_priority, find_output,
+			old_output->name);
+	if (old_index < 0) {
+		return;
+	}
+
+	int new_index = list_seq_find(ws->output_priority, find_output,
+			output->name);
+	if (new_index < 0) {
+		list_insert(ws->output_priority, old_index, strdup(output->name));
+	} else if (new_index > old_index) {
+		char *name = ws->output_priority->items[new_index];
+		list_del(ws->output_priority, new_index);
```

Comment:
No, because instead of using `strdup(output->name)` like I do if it's not in the list, I'm just re-inserting the same copy on the following line. I could do the following, but I felt it was a waste to make a duplicate after freeing a duplicate:

````c
char *name = ws->output_priority->items[new_index];
list_del(ws->output_priority, new_index);
free(name);
list_insert(ws->output_priority, old_index, strdup(output->name));
```

Reply:
Ah, yep. I had a derp moment. It was 12:30am local time.